src/OpenTelemetry/Trace/Span.cs

```diff
@@ -101,9 +101,11 @@ public sealed class Span : ISpan
                 {
                     this.links = new List<Link>(this.tracerConfiguration.MaxNumberOfLinks);
 
-                    foreach (var link in links.Reverse().Take(this.tracerConfiguration.MaxNumberOfLinks))
+                    var parentLinks = links.ToList();
+                    var takeCount = Math.Min(this.tracerConfiguration.MaxNumberOfLinks, parentLinks.Count);
```

Comment:
```
if (parentLinks.Count < MaxNumberOfLinks)
  this.links = parentLinks; // common case - no allocations
else
 this.links = parentLinks.GetRange(parentLinks.Count - Max, Max);
```

Reply:
Right, we can just use their reference then.
Thanks I'll figure it out. 12 hours at work my brain is failing me now

Reply:
1. Being realistic and hoping one doesn't pass a list, then tries to reuse it.
2. TIL `GetRange`

Thanks a lot!